code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -2472,6 +2472,56 @@
 		T.health -= 20
 		T.mutation_mod += 0.1
 
+/datum/reagent/toxin/insecticide
+	name = "Insecticide"
+	id = INSECTICIDE
+	description = "A broad pesticide. Do not ingest!"
+	reagent_state = REAGENT_STATE_LIQUID
+	color = "#49002E" //rgb: 73, 0, 46
+	density = 1.08
+	specheatcap = 4.18
+
+/datum/reagent/toxin/insecticide/reaction_mob(var/mob/living/M, var/method = TOUCH, var/volume)
+
+	if(..())
+		return 1
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		if(!C.wear_mask) //If not wearing a mask
+			C.adjustToxLoss(REM) //4 toxic damage per application, doubled for some reason
+		if(isinsectoid(C) || istype(C, /mob/living/carbon/monkey/roach)) //Insecticide being poisonous to bugmen, who'd've thunk
+			M.adjustToxLoss(10 * REM)
+					
+/datum/reagent/toxin/insecticide/reaction_animal(var/mob/living/simple_animal/M, var/method=TOUCH, var/volume)
+	if(..())
+		return 1
+		
+	if(istype(M,/mob/living/simple_animal/hostile/bigroach))
+		M.gib()
```

Comment:
This has a lot of issues with it.

1. This is really not OOP. This should go inside `reagent_act()` on the associated mob
2. Even if you were going to do it like this, you should've made a list of types and then done `is_type_in_list()` once instead of a series of if statements.
3. This isn't consistent with toxins, which just instantly kills them without gibbing, which makes much more sense for insecticide anyway. This is the only way to kill roaches and such but still leave behind a body, which matters for some applications like grinding roaches for shells.

In short: delete all of this and just add to or replace toxin in their respective `reagent_act()` procs.

Reply:
I am a codelet and literally jury rigged this to the best of my understanding, please give me steps to do or the specific changes.

Reply:
go to each mob's file (e.g. `simple_animal/roach.dm`) and look for a proc called `reagent_act()` (L292 in the case of roach.dm) (ctrl L to toggle line numbers, ctrl G to go to line).  In the case of roaches, your job is 98% already done, just add `, INSECTICIDE` after `TOXIN` at line 299.

Repeat this inside the reagent_act() proc for bees and spiderlings, then delete this whole series of if statements.

Reply:
Done.